infra/scripts/replace-running-components.sh

```diff
@@ -0,0 +1,173 @@
+        echo "VIC_NAME: name of VCH; matches --name argument for vic-machine"
+        echo "VIC_ID: ID of VCH, as displayed in output of vic-machine ls"
+        if [[ $interactive -eq 0 ]]; then
+            exit 1
+        fi
+        read -p "Or enter VCH name to continue: " VIC_NAME
+    fi
+}
+
+# Falls back on VIC_ID if possible, bails if not, assumes the VCH name is ambiguous
+function check-name-isnt-ambiguous ()  {
+    username=$(govc env | grep GOVC_USERNAME | cut -d= -f2)
+    password=$(govc env | grep GOVC_PASSWORD | cut -d= -f2)
+    target=$(govc env | grep GOVC_URL | cut -d= -f2)
+
+    if [[ ! -v VIC_ID ]] && [[ $(vch-name-is-ambiguous) ]]; then
+        echo "The provided VIC name is ambiguous; please choose the correct VCH ID from the output below and assign it to the environment variable VIC_ID, e.g., export VIC_ID=12"
+
+        $VIC_DIR/bin/vic-machine-linux\
+            ls \
+            --target $GOVC_URL \
+            --user $GOVC_USER \
+            --password=$GOVC_PASSWORD \
+            --thumbprint=$(get-thumbprint)
+
+        if [[ $interactive -eq 0 ]]; then
+            exit 1
+        fi
+        read -p "Enter unique VIC ID from above: " VIC_ID
+    fi
+}
+
+# Translates VCH name to ID if necessary
+function get-vic-id () {
+    if [[ -z $VIC_ID ]]; then
+        export VIC_ID="$($VIC_DIR/bin/vic-machine-linux ls --target=$target --user=$username --password=$password --thumbprint=$(get-thumbprint) | grep $VIC_NAME | awk '{print $1}')"
+    fi
+}
+
+function get-ssh-keys() {
+    if [[ -z $VIC_KEY ]]; then
+        echo "Variable VIC_KEY not set. Provide the path to your public SSH key below."
+        if [[ $interactive -eq 0 ]]; then
+            exit 1
+        fi
+        read -p "Path to your public SSH key for access to VCH [/home/$USER/.ssh/id_rsa.pub]: " key
+        export VIC_KEY=${key:-/home/$USER/.ssh/id_rsa.pub}
+    fi
+}
+
+# Checks environment for required inputs
+function sanity-checks () {
+    check-govc-vars
+    check-vch-name-or-id
+    check-name-isnt-ambiguous
+    get-vic-id
+    get-ssh-keys
+}
+
+# Enables SSH and saves off the VCH IP address
+function enable-debug () {
+    VCH_IP=$($VIC_DIR/bin/vic-machine-linux debug \
+                                   --target=$target \
+                                   --id=$VIC_ID \
+                                   --user=$username \
+                                   --password=$password \
+                                   --authorized-key=$VIC_KEY \
+                                   --thumbprint=$(get-thumbprint) \
+                 | grep -A1 "Published ports" | tail -n1 | awk '{print $NF}')
+}
+
+function replace-component() {
+    scp -oUserKnownHostsFile=/dev/null -oStrictHostKeyChecking=no $VIC_DIR/bin/$1 root@$VCH_IP:/tmp/$1 2>/dev/null
+    on-vch mv /sbin/$1 /tmp/old-$1
+    on-vch mv /tmp/$1 /sbin/$1
+    on-vch chmod 755 /sbin/$1
```

Comment:
suggest doing this first before the `mv /sbin/$1 /tmp/old-$1` - that way there's no window in which vic-init will try and fail to execute it.